tests/publiccloud/migration.pm

```diff
@@ -26,7 +26,10 @@ sub run {
     $self->select_serial_terminal;
     my $provider = $self->provider_factory();
     my $instance = $provider->create_instance();
-    $instance->wait_for_guestregister();
+    $instance->wait_for_guestregister() if is_ondemand();
```

Comment:
Why are we only running this on ondemand images? The routine is expected to work on all instance types IIRC.

Reply:
for DMS we choose only BYOS or ondemand only

Reply:
look at this description: https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/instance.pm#L170
```
In case of BYOS images we checking that service is inactive and quit
Returns the time needed to wait for the guestregister to complete.
```
so it should work also in BYOS.